pageserver/src/tenant/timeline.rs

```diff
@@ -3372,23 +3032,19 @@ impl Timeline {
 
         let mut all_keys = Vec::new();
 
-        let downcast_deltas: Vec<_> = deltas_to_compact
-            .iter()
-            .map(|l| l.clone().downcast_delta_layer().expect("delta layer"))
-            .collect();
-        for dl in downcast_deltas.iter() {
-            // TODO: replace this with an await once we fully go async
-            all_keys.extend(DeltaLayer::load_keys(dl, ctx).await?);
+        for l in deltas_to_compact.iter() {
+            all_keys.extend(l.load_keys(ctx).await?);
         }
 
+        // FIXME: should spawn_blocking the rest of this function
```

Comment:
Why? If you want to commit this FIXME, then
1. explain why (and why it's ok to punt on it for now)
2. create follow-up issue and link to it from here

Reply:
It's listed as a follow-up on the PR. I don't think an issue is needed for this either.